tests/storage/migration.go

```diff
@@ -846,65 +849,214 @@ var _ = SIGDescribe("Volumes update with migration", decorators.RequiresTwoSched
+					libdv.WithBlankImageSource(),
+					libdv.WithStorage(libdv.StorageWithStorageClass(sc),
+						libdv.StorageWithVolumeSize("1G"),
+						libdv.StorageWithVolumeMode(volumeMode),
+						libdv.StorageWithAccessMode(k8sv1.ReadWriteOnce),
+					),
+				)
+				_, err = virtClient.CdiClient().CdiV1beta1().DataVolumes(ns).Create(context.Background(),
+					hpDV, metav1.CreateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				addVolume(vm.Name, vm.Namespace, hpVolName, hpDV.Name)
+				createFileOnHotpluggedVol(vmi, hpVolName)
+
+				indexRoot := getIndexVol(vm.Name, vm.Namespace, rootVolName)
+				indexHp := getIndexVol(vm.Name, vm.Namespace, hpVolName)
+
+				if dstBlock {
+					sc, exist = libstorage.GetRWOBlockStorageClass()
+					volumeMode = k8sv1.PersistentVolumeBlock
+					Expect(exist).To(BeTrue())
+				} else {
+					sc = testSc
+					volumeMode = k8sv1.PersistentVolumeFilesystem
+				}
+				dvRootDst := libdv.NewDataVolume(
+					libdv.WithBlankImageSource(),
+					libdv.WithStorage(libdv.StorageWithStorageClass(sc),
+						libdv.StorageWithVolumeSize("2Gi"),
+						libdv.StorageWithVolumeMode(volumeMode),
+						libdv.StorageWithAccessMode(k8sv1.ReadWriteOnce),
+					),
+				)
+				_, err = virtClient.CdiClient().CdiV1beta1().DataVolumes(ns).Create(context.Background(),
+					dvRootDst, metav1.CreateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				dvHpDst := libdv.NewDataVolume(
+					libdv.WithBlankImageSource(),
+					libdv.WithStorage(libdv.StorageWithStorageClass(sc),
+						libdv.StorageWithVolumeSize("2Gi"),
+						libdv.StorageWithVolumeMode(volumeMode),
+						libdv.StorageWithAccessMode(k8sv1.ReadWriteOnce),
+					),
+				)
+				_, err = virtClient.CdiClient().CdiV1beta1().DataVolumes(ns).Create(context.Background(),
+					dvHpDst, metav1.CreateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				p, err := patch.New(
+					patch.WithReplace("/spec/dataVolumeTemplates", []virtv1.DataVolumeTemplateSpec{}),
+					patch.WithReplace(fmt.Sprintf("/spec/template/spec/volumes/%d/dataVolume/name", indexRoot), dvRootDst.Name),
+					patch.WithReplace(fmt.Sprintf("/spec/template/spec/volumes/%d/dataVolume/name", indexHp), dvHpDst.Name),
+					patch.WithReplace("/spec/updateVolumesStrategy", virtv1.UpdateVolumesStrategyMigration),
+				).GeneratePayload()
+				Expect(err).ToNot(HaveOccurred())
+
+				By(fmt.Sprintf("Update root DV %s with %s and hotplug dv %s with %s", rootDV.Name, dvRootDst.Name, hpDV.Name, dvHpDst.Name))
+				vm, err = virtClient.VirtualMachine(vm.Namespace).Patch(context.Background(), vm.Name, types.JSONPatchType, p, metav1.PatchOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				waitForMigrationToSucceed(virtClient, vm.Name, vm.Namespace)
+				checkFileOnHotpluggedVol(vmi)
+			},
+				Entry("from filesystem to filesystem", false, false),
+				Entry("from filesystem to block", false, true),
+				Entry("from block to filesystem", true, false),
+				Entry("from block to block", true, true),
+			)
```

Comment:
A little worried about the amount of test additions here. Any chance we could unit test some of the scenarios?

If it's absolutely a must, is https://github.com/kubevirt/kubevirt/pull/13725 blocked or could we get that in to help with at least parallelizing?